File: LeetCode/Medium/0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Deque<Integer> stack = new ArrayDeque<>();
        int[] answer = new int[temperatures.length];

        for (int i = 0; i < temperatures.length; i++) {
            while (!stack.isEmpty() && temperatures[stack.peek()] < temperatures[i]) {
                int top = stack.pop();
                answer[top] = i - top;
            }
            stack.push(i);
        }

        return answer;
    }
}

File: LeetCode/Medium/0739-daily-temperatures/0739-daily-temperatures.kt
class Solution {
    fun dailyTemperatures(temperatures: IntArray): IntArray {
        val stack = ArrayDeque<Int>()
        val answer = IntArray(temperatures.size)

        for (i in temperatures.indices) {
            while (stack.isNotEmpty() && temperatures[stack.first()] < temperatures[i]) {
                val top = stack.removeFirst()
                answer[top] = i - top
            }
            stack.addFirst(i)
        }

        return answer
    }
}